python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -133,22 +147,36 @@ def runtimeEvalString(this_obj, attr_name, val):
 
 
 def raw_eval(val):
+    ## Attempts to evaluate the val object and return the object it evaluates to if it is a Python object
+    ## Makes use of basic caching as we don't expect that things at this level should change.
+    from copy import deepcopy
+    if val in _eval_cache:
+        return deepcopy(_eval_cache[val])
+
     try:
+        ## Catch common strings which shouldn't be evaluated anyway and return them asif an exception occured
+        if val in ['', '.', 'new']:
+            _eval_cache[val] = val
+            return deepcopy(val)
```

Comment:
OK so here's a gotcha:

```File.name``` defaults to ```''```. This is saved in the Schema as a SimpleItem.

Ganga attempts to work out what the 'type' of ```''``` is but ```''``` could also be ```'myClass'```, due to the way that we allowed strings to be parsed (at least in the past).
As a result Ganga has to try and evaluate the object and work out if it is the latter or the former.
If the ```defvalue``` is the latter we do one thing and if the ```defvalue``` is ```''``` then we do another.

If we intend to keep supporting things the way they are atm we don't throw an exception. If we do then we break a lot of stuff, and frankly that was not the intention of this PR. I'm happy to remove the ```''``` special case but fixing every last bit of the schema here isn't what I wanted to do, nor what the PR set out to accomplish.